homeassistant/components/shopping_list.py

```diff
@@ -175,23 +211,32 @@ def async_update(self, item_id, info):
         return item
 
     @callback
-    def async_clear_completed(self):
+    def async_clear_completed(self, list_id):
         """Clear completed items."""
-        self.items = [itm for itm in self.items if not itm['complete']]
+        lis = next((li for li in self.lists if li['id'] == list_id), None)
+
+        if lis is None:
+            raise KeyError
+
+        lis['items'] = [itm for itm in lis['items'] if not itm['complete']]
         self.hass.async_add_job(self.save)
 
     @asyncio.coroutine
     def async_load(self):
         """Load items."""
         def load():
             """Load the items synchronously."""
-            return load_json(self.hass.config.path(PERSISTENCE), default=[])
+            return load_json(self.hass.config.path(PERSISTENCE), default=[{
```

Comment:
If we're migrating to multiple lists, we should also migrate to the Storage helper. 

Here is an example of how to migrate existing files to the storage helper https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity_registry.py#L206-L210

Reply:
Is this so it will be stored in `.storage`? Agreed. Thanks.

Reply:
Yep

Reply:
I'll try to figure this out tonight.